Index: src/qp/optimizer/RandomOptimizer.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/qp/optimizer/RandomOptimizer.java	(date 1553936750755)
+++ src/qp/optimizer/RandomOptimizer.java	(date 1553936750755)
@@ -435,6 +435,8 @@
 	    return findNodeAt(((Select)node).getBase(),joinNum);
 	}else if(node.getOpType()==OpType.PROJECT){
 	    return findNodeAt(((Project)node).getBase(),joinNum);
+	}else if(node.getOpType()==OpType.DISTINCT){
+		return findNodeAt(((Distinct)node).getBase(),joinNum);
 	}else{
 	    return null;
 	}
@@ -462,6 +464,11 @@
 	    modifySchema(base);
 	    Vector attrlist = ((Project)node).getProjAttr();
 	    node.setSchema(base.getSchema().subSchema(attrlist));
+	}else if(node.getOpType()==OpType.DISTINCT){
+		Operator base = ((Distinct)node).getBase();
+		modifySchema(base);
+		Vector attrlist = ((Distinct)node).getProjAttr();
+		node.setSchema(base.getSchema().subSchema(attrlist));
 	}
     }
 
@@ -521,6 +528,10 @@
 	    Operator base = makeExecPlan(((Project)node).getBase());
 	    ((Project)node).setBase(base);
 	    return node;
+	}else if(node.getOpType() == OpType.DISTINCT){
+		Operator base = makeExecPlan(((Distinct)node).getBase());
+		((Distinct)node).setBase(base);
+		return node;
 	}else{
 	    return node;
 	}
